openassessment/xblock/static/js/src/lms/oa_staff_area.js

```diff
@@ -310,30 +372,72 @@
                 // final grade and in the future should include details of
                 // the staff override itself.
                 view.loadStudentInfo({expanded_view: 'final-grade'});
-            }).fail(function(errorMessage) {
-                $('.staff-override-error').html(_.escape(errorMessage));
-                view.staffSubmitEnabled(true);
+            };
+            this.callStaffAssess(submissionID, rubric, scope, successCallback, '.staff-override-error');
+        },
+
+        /**
+         * Submit the staff grade, and check out another learner for grading if continueGrading is true.
+         *
+         * @param {string} submissionID The ID of the submission to be submitted.
+         * @param {element} rubric The rubric element to be assessed.
+         * @param {element} scope An ancestor element for the submit button (to allow for shared
+         *     classes in different form).
+         * @param {boolean} continueGrading If true, another learner will be marked as "In Progress",
+         *     and a new grading form will be rendered with the learner's answer.
+         */
+        submitStaffGrade: function(submissionID, rubric, scope, continueGrading) {
+            var view = this;
+            var successCallback = function() {
+                view.staffGradeFormLoaded = false;
+                var showFullGradeTab = function() {
+                    // Need to show the staff grade component again, unfortunately requiring a global selector.
+                    $('.button-staff-grading').click();
+                    if (continueGrading) {
+                        $('.staff__grade__title', view.element).click();
+                    }
+                };
+                view.load(showFullGradeTab);
+            };
+            this.callStaffAssess(submissionID, rubric, scope, successCallback, '.staff-grade-error');
+        },
+
+        /**
+         * Make the server call to submit the staff assessment.
+         *
+         * @param {string} submissionID The ID of the submission to be submitted.
+         * @param {element} rubric The rubric element to be assessed.
+         * @param {element} scope An ancestor element for the submit button (to allow for shared
+         *     classes in different form).
+         * @param {function} successCallback A function to execute on success.
+         * @param {string} errorSelector a CSS class selector for displaying error messages.
+         */
+        callStaffAssess: function(submissionID, rubric, scope, successCallback, errorSelector) {
+            var view = this;
+            view.staffSubmitEnabled(scope, false);
+
+            this.server.staffAssess(
+                rubric.optionsSelected(), rubric.criterionFeedback(), rubric.overallFeedback(), submissionID
+            ).done(successCallback).fail(function(errorMessage) {
+                // TODO: this needs styling help Andy!
+                scope.find(errorSelector).html(_.escape(errorMessage));
```

Comment:
@andy-armstrong The display of the error messages all need some styling help (more padding).